scripts/update_pwnagotchi.sh

```diff
@@ -36,70 +37,79 @@ function test_root() {
 function test_github() {
     wget -q  --spider $GIT_URL
     if [ $? -ne 0 ]; then
-        echo "[!] Cannot reach github. This script requires internet access, ensure connection sharing is working."
+        echo "[!] Cannot reach github. This script requires internet access and DNS resolution, ensure connection sharing is working and valid DNS server in /etc/resolv.conf."
         exit 2
     fi
 }
-
-echo "[+] Checking prerequisites."
-test_root
-test_github
-
-while getopts ":v:u:m:b:r:h" o; do
+while getopts "v:u:m:brh" o; do
   case "${o}" in
     v)
       VERSION="${OPTARG}"
       ;;
     u)
       GIT_URL="${OPTARG}"
       ;;
-    m)
-      if [[ "${SUPPORTED_RESTART_MODES[*]}" =~ ${OPTARG} ]]; then
-        MODE="${OPTARG}"
-      else
-        usage
-      fi      
-      ;;      
     b)
       BACKUPCONFIG=1
       ;;
     r)
-      BACKUPCONFIG=1    
+      BACKUPCONFIG=1
       RESTORECONFIG=1
-      ;;      
+      ;;
     h)
       usage
       ;;
+    m)
+      if [[ "${SUPPORTED_RESTART_MODES[*]}" =~ ${OPTARG} ]]; then
+        MODE="${OPTARG}"
+      else
+        usage
+      fi
+      ;;
     *)
       usage
       ;;
   esac
 done
 shift $((OPTIND-1))
 
+echo "[+] Checking prerequisites."
+test_root
+test_github
+
 # clean up old files, clone master, set checkout to commit if needed.
 echo "[+] Cloning to $GIT_FOLDER..."
-rm $GIT_FOLDER -rf
```

Comment:
I added this here before the clone to ensure git would not throw a fit about the path being in use.

Reply:
doh, missed that from debugging. fixed in latest push